File: Go/maximize-happiness-of-selected-children.go
func maximumHappinessSum(happiness []int, k int) int64 {
    slices.Sort(happiness)
    total := int64(0)
    n := len(happiness)
    for i := 0; i < len(happiness) && i < k; i++ {
        total += int64(max(0, happiness[n - i - 1] - i))
    }
    return total
}

File: TypeScript/maximize-happiness-of-selected-children.ts
function maximumHappinessSum(happiness: number[], k: number): number {
    happiness.sort((a, b) => b - a);
    let total = 0;
    for (let i = 0; i < happiness.length && i < k; i++) {
        total += Math.max(0, happiness[i] - i);
    }
    return total;
};
